lib/src/http.ts

```diff
@@ -28,6 +28,7 @@ export function isPathParamTypeSafe(type: Type, typeTable: TypeTable): boolean {
     case TypeKind.ARRAY:
       return isParamArrayElementTypeSafe(type.elementType, typeTable);
     case TypeKind.UNION:
+    case TypeKind.INTERSECTION:
```

Comment:
Yeah, I actually hit this because we have one of our APIs where we have two params say:
`v1/accounts/:accountId/purchase/:purchaseId`

So it was defined as `AccountId & PurchaseId`

Reply:
I'm not sure I follow, your example would be written similar to:

```ts
@endpoint({
  method: "GET",
  path: "/v1/accounts/:accountId/purchase/:purchaseId"
})
class Endpoint {
  @request
  request(
    @pathParams
    pathParams: AccountId & PurchaseId
  ) {}
}

interface AccountId {
  accountId: String;
}

interface PurchaseId {
  purchaseId: String;
}
```

Neither `accountId` nor `purchaseId` types would require the use of intersection type.

Reply:
The intersection type here just attempts to be a composite data type, allowing us to assign both `AccountId` and `PurchaseId` to the `pathParams` object. 
For example:
```
@endpoint({
  method: "GET",
  path: "/v1/accounts/:accountId/purchase/:purchaseId"
})
class Endpoint {
  @request
  request(
    @pathParams
    pathParams: EndPointParams
  ) {}
}

interface AccountId {
  accountId: String;
}

interface PurchaseId {
  purchaseId: String;
}

type EndPointParams = AccountId & PurchaseId
```

Allows it to be cleaner and also allows us to reuse `AccountId` or `PurchaseId` when we see fit.

Reply:
Ah I see, I didn't think of that use case 👍